File: proj.android_studio/ee-x-store/src/main/java/com/ee/store/StoreException.java
package com.ee.store;

import com.android.billingclient.api.BillingClient.BillingResponseCode;

public final class StoreException extends RuntimeException {
    @BillingResponseCode
    public final int responseCode;

    StoreException(@BillingResponseCode int responseCode) {
        super("Error during store operation. Response code: " + responseCode);
        this.responseCode = responseCode;
    }
}

File: proj.android_studio/ee-x-store/src/main/java/com/ee/store/StoreException.kt
package com.ee.store

import com.android.billingclient.api.BillingClient.BillingResponseCode

internal class StoreException constructor(
    @field:BillingResponseCode @param:BillingResponseCode val responseCode: Int)
    : RuntimeException("Error during store operation. Response code: $responseCode")